src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2021 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.migration.db;
+
+import io.dropwizard.hibernate.AbstractDAO;
+import nl.knaw.dans.migration.core.tables.EasyFile;
+import nl.knaw.dans.migration.core.tables.ExpectedDataset;
+import nl.knaw.dans.migration.core.tables.InputDataset;
+import org.hibernate.SessionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class InputDatasetDAO extends AbstractDAO<EasyFile> {
+  private static final Logger log = LoggerFactory.getLogger(InputDatasetDAO.class);
+
+  public InputDatasetDAO(SessionFactory sessionFactory) {
+    super(sessionFactory);
+  }
+
+  public void create(InputDataset input) {
+    log.trace(input.toString());
+    currentSession().save(input);
+  }
+
+  public void deleteByDoi(String doi, String source) {
+    log.trace("deleting InputDataset {} {}", doi, source);
+    int r = currentSession()
+        .createQuery("DELETE FROM InputDataset WHERE doi = :doi AND source = :source")
```

Comment:
Can the source check cause trouble for load-from-vault when you rerun with a "fixed" store-name? I added the check suspecting datasets might be in fedora as well as in the bag-store.

Reply:
that's a sensible check, considering the landingpage-only pages we are adding now are taken from Fedora, even from Vault datasets

Reply:
actually, i would also like to add the batch to this check. So only the current file is reloaded

Reply:
actually, i would also like to add the batch to this check. So only the current file is reloaded